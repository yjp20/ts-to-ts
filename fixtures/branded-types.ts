/** @model */
type EmailAddress = string & { __brand: "email" };

/** @model */
type UserId = number & { __brand: "userId" };

/** @model */
type PositiveNumber = number & { __brand: "positive" };

/** @model */
interface ValidatedUser {
  email: EmailAddress;
  id: UserId;
  score: PositiveNumber;
}
