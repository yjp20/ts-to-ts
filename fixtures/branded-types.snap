<<<<<<<<<<< typescript <<<<<<<<<<<<
/** @model */
type EmailAddress = string & { __brand: "email" };

/** @model */
type UserId = number & { __brand: "userId" };

/** @model */
type PositiveNumber = number & { __brand: "positive" };

/** @model */
interface ValidatedUser {
  email: EmailAddress;
  id: UserId;
  score: PositiveNumber;
}
===================================
// branded-types.ts
alias EmailAddress = {
  toString: {};
  charAt: {};
  charCodeAt: {};
  concat: {};
  indexOf: {};
  lastIndexOf: {};
  localeCompare: {};
  match: {};
  replace: {};
  search: {};
  slice: {};
  split: {};
  substring: {};
  toLowerCase: {};
  toLocaleLowerCase: {};
  toUpperCase: {};
  toLocaleUpperCase: {};
  trim: {};
  length: float64;
  substr: {};
  valueOf: {};
  __brand: "email";
};

alias UserId = {
  toString: {};
  toFixed: {};
  toExponential: {};
  toPrecision: {};
  valueOf: {};
  toLocaleString: {};
  __brand: "userId";
};

alias PositiveNumber = {
  toString: {};
  toFixed: {};
  toExponential: {};
  toPrecision: {};
  valueOf: {};
  toLocaleString: {};
  __brand: "positive";
};

model ValidatedUser {
  email: EmailAddress;
  id: UserId;
  score: PositiveNumber;
};
>>>>>>>>>>> typespec >>>>>>>>>>>>>>